Release notes for Red Hat Developer Hub 1.6.0 (#1133)
redhat-developer/red-hat-developers-documentation-rhdh: modules/release-notes/ref-release-notes-breaking-changes.adoc

This section lists breaking changes in {product} {product-version}.

> [id="removed-functionality-rhidp-6215"]
> == The Topology-specific permission `topology.view.read` is removed
> 
> Previously, the Topology plugin used `topology.view.read` permission to control access. Users were unable to configure Topology permissions using the RBAC UI. With this update, users can configure Kubernetes plugin permissions using the RBAC UI, which now governs the access to the Topology plugin. You can now use Kubernetes plugin permissions `kubernetes.clusters.read` and `kubernetes.resources.read` for the Topology plugin, as the topology-specific permission `topology.view.read` is removed. 
> 
> If you are using a CSV permission file, update the following line:
> 
> .Old topology permission definition
> [source,csv]
> ----
> p, role:default/topology-viewer, topology.view.read, read, allow
> ----
> 
> .New topology permission definition
> [source,csv]
> ----
> p, role:default/topology-viewer, kubernetes.clusters.read, read, allow
> p, role:default/topology-viewer, kubernetes.resources.read, read, allow
> ----
> 
> 
> .Additional resources
> * link:https://issues.redhat.com/browse/RHIDP-6215[RHIDP-6215]

Reviewer: ```suggestion
[id="removed-functionality-rhidp-6215"]
== The Topology-specific permission `topology.view.read` is removed

Previously, the Topology plugin used `topology.view.read` permission to control access. Users were unable to configure Topology permissions using the RBAC UI. With this update, users can configure Kubernetes plugin permissions using the RBAC UI, which now governs the access to the Topology plugin. You can now use Kubernetes plugin permissions `kubernetes.clusters.read`, `kubernetes.resources.read` and `kubernetes.proxy` for the Topology plugin, as the Topology-specific permission `topology.view.read` is removed.

If you are using a CSV permission file, update the following lines:

.Old Topology permission definition
[source,csv]
----
p, role:default/topology-viewer, topology.view.read, read, allow
p, role:default/topology-viewer, kubernetes.proxy, use, allow
----

.New Topology permission definition
[source,csv]
----
p, role:default/topology-viewer, kubernetes.clusters.read, read, allow
p, role:default/topology-viewer, kubernetes.resources.read, read, allow
p, role:default/topology-viewer, kubernetes.proxy, use, allow
----

.Additional resources
* link:https://issues.redhat.com/browse/RHIDP-6215[RHIDP-6215]

[id="removed-functionality-rhtapui-13"]
== The Tekton-specific permission `tekton.view.read` is removed

Previously, the Tekton plugin used `tekton.view.read` permission to control access. Users were unable to configure Tekton permissions using the RBAC UI. With this update, users can configure Kubernetes plugin permissions using the RBAC UI, which now governs the access to the Tekton plugin. You can now use Kubernetes plugin permissions `kubernetes.clusters.read`, `kubernetes.resources.read` and `kubernetes.proxy` for the Tekton plugin, as the Tekton-specific permission `tekton.view.read` is removed. 

If you are using a CSV permission file, update the following lines:

.Old Tekton permission definition
[source,csv]
----
p, role:default/tekton-viewer, tekton.view.read, read, allow
p, role:default/tekton-viewer, kubernetes.proxy, use, allow
----

.New Tekton permission definition
[source,csv]
----
p, role:default/tekton-viewer, kubernetes.clusters.read, read, allow
p, role:default/tekton-viewer, kubernetes.resources.read, read, allow
p, role:default/tekton-viewer, kubernetes.proxy, use, allow
----

.Additional resources
* link:https://issues.redhat.com/browse/RHTAPUI-13[RHTAPUI-13]
```

[id="removed-functionality-rhidp-7365"]
== Migration to the core Auditor service

> The Auditor format, including audit fields and event names, and IDs, has been updated to align with the new Auditor service conventions. Filtering queries based on the old format may fail to work as expected.

Reviewer 2: I think when you mention the Auditor service conventions, it would be helpful to clarify that they now come from the upstream project: https://backstage.io/docs/backend-system/core-services/auditor/

.Additional resources
* link:https://issues.redhat.com/browse/RHIDP-7365[RHIDP-7365]

[id="removed-functionality-rhidp-7373"]
== {product} introduces the Backstage Audit Log Service

> {product} {product-version} introduces the Backstage Audit Log Service, which replaces the custom audit logging system. This a significant structural and behavioral change to how audit events are generated and consumed.

Reviewer 2: ```suggestion
{product} {product-version} introduces the Backstage Audit Log Service, which replaces the custom audit logging system. This is a significant structural and behavioral change to how audit events are generated and consumed.

```

Reviewer 3: reminder that the RN script will apply typo corrections when you run the script again after updating the JIRA RN text fields. So no need to comment here -- just fix the typos in https://issues.redhat.com/browse/RHIDP-7373 directly.

> The key changes following this transition include::

Reviewer 2: ```suggestion
The key changes following this transition include:

```
